### PR TITLE
EXL-298 - Imported Movements from iService2

### PR DESCRIPTION
**Pull request by zsso**

### CODE REVIEW

File: plugin/hook.php [hunk 1]
```diff
@@ -66,16 +66,23 @@ function plugin_iservice_redefine_menus($menus): array
     return RedefineMenus::redefine($menus);
 }
 
-function plugin_iservice_PluginFieldsTicketticketcustomfield_update(PluginFieldsTicketticketcustomfield $item)
+function plugin_iservice_ticket_add(Ticket $item)
```
Comment (hupupu) [inline plugin/hook.php:69]: Does this work in lowercase? Shouldn't it be `plugin_iservice_Ticket_add`

File: plugin/inc/movement.class.php [hunk 2]
```diff
@@ -13,7 +13,7 @@ class PluginIserviceMovement extends CommonDBTM
     const TYPE_MOVE = 'move';
 
     public static $rightname      = 'plugin_iservice_movement';
-    public static $expert_line_id = 525;
+    public static $expert_line_id = 506; // TODO: confirm with hupu!
```
Comment (hupupu) [inline plugin/inc/movement.class.php:16]: I think we created a function for this which gets the Id of a record based on it's name, didn't we?

File: plugin/locales/ro_RO.po
Comment (hupupu) [inline plugin/locales/ro_RO.po:1]: Shouldn't we create the entries at least in the English file?

**Review by hupupu — APPROVED** [review]

File: plugin/hook.php (hunk 1 above)
Comment (zsso) [reply plugin/hook.php:69]: I test it and looks like the `plugin_iservice_Ticket_add` function is not executed regardless of casing. However the `plugin_iservice_ticket_update` method is, even on ticket creation, so maybe we don't need `plugin_iservice_Ticket_add` at all.

File: plugin/inc/movement.class.php (hunk 2 above)
Comment (zsso) [reply plugin/inc/movement.class.php:16]: I can't find anything like that. As I remember I wanted to do something similar for "aviz" and "factura", but finally we used a const instead.

File: plugin/locales/ro_RO.po
Comment (zsso) [reply plugin/locales/ro_RO.po:1]: You mean that I forgot to compile translations to .mo?

File: plugin/inc/utils/toolbox.class.php [hunk 3]
```diff
@@ -306,4 +306,23 @@ public static function getCodmatValue(string $codmat): string
...
+            return null;
+        }
+
+        if (empty($attributeToSearchFor)) {
+            $attributeToSearchFor = $modelName::getNameField();
+        }
+
+        $queryResult = PluginIserviceDB::getQueryResult("SELECT $attributeToReturn FROM $table WHERE $attributeToSearchFor = '$attributeValue'");
+
+        if (count($queryResult) > 1) {
```
Comment (hupupu) [inline plugin/inc/utils/toolbox.class.php:321]: what if there is no result?
```suggestion
        if (count($queryResult) != 1) {
```

File: plugin/inc/utils/toolbox.class.php [hunk 4]
```diff
@@ -306,4 +306,23 @@ public static function getCodmatValue(string $codmat): string
         return self::$codmatValues[$codmat]['denum'] ?? '';
     }
 
+    public static function getIdentifierAttributeByAttribute(string $modelName, string $attributeValue, string $attributeToSearchFor = null, string $attributeToReturn = 'id'): mixed
```
Comment (hupupu) [inline plugin/inc/utils/toolbox.class.php:309]: (optional)
```suggestion
    public static function getIdentifierByAttribute(string $modelName, string $attributeValue, string $attributeToSearchFor = null, string $idAttributeName = 'id'): mixed
```

File: plugin/inc/utils/toolbox.class.php [hunk 5]
```diff
@@ -306,4 +306,23 @@ public static function getCodmatValue(string $codmat): string
...
+    {
+        if (!($table = getTableForItemType($modelName))) {
+            return null;
+        }
+
+        if (empty($attributeToSearchFor)) {
+            $attributeToSearchFor = $modelName::getNameField();
+        }
+
+        $queryResult = PluginIserviceDB::getQueryResult("SELECT $attributeToReturn FROM $table WHERE $attributeToSearchFor = '$attributeValue'");
```
Comment (hupupu) [inline plugin/inc/utils/toolbox.class.php:319]: (optional)
Can't we use a model's find function?

**Review by hupupu — APPROVED** [review]

File: plugin/inc/utils/toolbox.class.php (hunk 5 above)
Comment (zsso) [reply plugin/inc/utils/toolbox.class.php:319]: findByCrit ??

**Pull request merged**